lib/Driver/Job.cpp

```diff
@@ -131,6 +131,16 @@ StringRef CommandOutput::getAdditionalOutputForType(types::ID Type) const {
   return getOutputForInputAndType(Inputs[0].Primary, Type);
 }
 
+SmallVector<StringRef, 16>
+CommandOutput::getAdditionalOutputsForType(types::ID Type) const {
+  SmallVector<StringRef, 16> V;
+  for (auto const &I : Inputs) {
+    V.push_back(getOutputForInputAndType(I.Primary, Type));
+  }
+  assert(V.empty() || V.size() == Inputs.size());
```

Comment:
Does this do the correct thing in WMO mode? Or is it just only being called from batch jobs at the moment?

Reply:
Only being called in batch.

Reply:
Maybe leave a comment to that effect. "Note: this function will fail if used from a normal CompileJob performing a SingleCompile."